RFCs/FS-1061-Allow-intrinsic-type-extensions-for-provided-types.md

```diff
@@ -0,0 +1,201 @@
+
+# Detailed design
+[design]: #detailed-design
+
+The compiler needs to be modified so that instead of an error being generated when a provided type is extended the type is extended:
+
+```
+Error FS0192: internal error: FindTypeDefBuilder: <Method> not found (FS0192) (<Application>)
+```
+
+Instead the type is extended in the same way as a normal type extension.
+
+The proposed systax is the same as the existing intrinsic type extension syntax:
+
+Example code short form:
+
+```fsharp
+type t = BasicProvider.Generative.MyType with
+    member x.Foo = 42
+```
+
+Example code long form:
+
+```fsharp
+type t = BasicProvider.Generative.MyType
+
+type t with 
+    member x.Foo = 42
+```
+
+Given that `MyType` is just a plain type with a default constructor the resulting generated code wold look like this:
+```
+type t =
+  class
+    new : unit -> t
+    member Foo : int
+```
+
+An addition change to language specification also needs to be changed so that an override can be applied as an intrinsic type extension too.   Imagine that BasicProvider.Generative.MyType has an abstract method as follows:
+
+```fsharp
+abstract member Rotate: float -> unit
+```
+
+It would be natural to describe this as follows:
+
+```fsharp
+type t = BasicProvider.Generative.MyType
+
+type t with 
+    member x.Foo = 42
+    override x.Rotate _ = ()
+```
+
+This would result in the following warning:
+```
+[FS0060] Override implementations in augmentations are now deprecated. Override implementations should be given as part of the initial declaration of a type.
+```
+
+The final part of this  proposal is to allow override implementations produced by generative type providers **only**, as it is not always possible to provide the implementation as part of the initial declaration of the type, especially if you are only the consumer of the type provider.  
+
+## Examples of current limitations that this RFC addresses
+
+As an example of the limitations imposed by not allowing intrinsic type extensions or override implementations on generative type providers I present the following examples.
+
+In the iOS designer type provider the usage looks like this:
+
+```fsharp
+type container = Xamarin.UIProvider 
+
+[<Register(container.myViewControllerBase.CustomClass)>] 
+type ViewController(handle : IntPtr) =  
+    inherit container.myViewControllerBase(handle)
+
+    override x.DidReceiveMemoryWarning() = 
+        base.DidReceiveMemoryWarning()
+        
+    override x.ViewDidLoad() =
+        x.DoNotPressAction <- Some (fun _ -> x.View.BackgroundColor <- UIColor.Purple)
+        base.ViewDidLoad()
+```
+
+This has several drawbacks that makes the user facing code more complex and unweidy than it should be with additions mentioned in the RFC.
+
+## Attributes have to be defined on outward facing type
```

Comment:
nit: can you use `###` for the header so that it's a sub-header of the limitations section?

Reply:
fixed